nevergrad/functions/images/core.py

```diff
@@ -20,7 +21,7 @@
 
 
 class Image(base.ExperimentFunction):
-    def __init__(self, problem_name: str = "recovering", index: int = 0) -> None:
+    def __init__(self, problem_name: str = "recovering", index: int = 0, loss=SumAbsoluteDifferencesLoss) -> None:
```

Comment:
It's better to just use a loss name here because that's easier to store as an information for the benchmarks.
If we can store it, we can then easily identify which were run with what kind of loss, so that we can compare. So for ExperimentFunction I strongly recommend keeping to str/int/bool/float only whenever possible

Reply:
Do you mean I should define loss='SumAbsoluteDifferences' and then have a series of if and elif to define the loss? Wouldn't that lead to code duplication and also require to update this function when adding a new loss making adding losses more tedious ?

Reply:
that's why it should be avoided if there are many losses, but used if there are not so many. Actually with only the current 2 you have now, I think you could remove 30 lines of code and it would work just as well + be more robust.
But if you plan on adding some as you seem to want, then let's keep something like this. Although I need to figure out how.

First thing I don't get is that is taking a reference and the other no, so both are incompatible. Are both losses supposed to work in both cases? I would be in favor of enforcing that an `ImageLoss` can accept a reference as input even if it does not use it. That removes some code. Then we type that `loss: ImageLoss` and we `self.add_descriptors(loss=loss.__name___)` since we need a string to describe it.

Reply:
I think you're right, it would make it better, the previous thing wasn't a good interface for all image lossses. I'll do what you suggest.